odl/test/discr/discr_ops_test.py

```diff
@@ -242,5 +235,27 @@ def test_resizing_op_adjoint(padding, fn_impl):
         assert almost_equal(inner1, inner2, places=dtype_places(dtype))
 
 
+def test_resizing_op_mixed_uni_nonuni():
+    """Check if resizing along uniform axes in mixed discretizations works."""
+    nonuni_part = odl.nonuniform_partition([0, 1, 4])
+    uni_part = odl.uniform_partition(-1, 1, 4)
+    part = uni_part.append(nonuni_part).append(uni_part).append(nonuni_part)
```

Comment:
Perhaps we should add an `extend` method that basically does

```
def extend(self, lst):
    result = self
    for el in lst:
        result = result.append(el)
    return result
```

Then you could do

```
part = uni_part.extend([nonuni_part, uni_part, nonuni_part])
```

which is much cleaner